src/stats/conversions.ts

```diff
@@ -209,32 +224,47 @@ function handleConversionCompute(
     return false;
   }
 
-  const opntInControl = isInControl(oppActionStateId);
   const opntDidLoseStock = prevOpponentFrame && didLoseStock(opponentFrame, prevOpponentFrame);
+  const playerDidLoseStock = prevPlayerFrame && didLoseStock(playerFrame, prevPlayerFrame);
+
+  const opntPosition = [opponentFrame.positionX, opponentFrame.positionY];
+  const opntIsOffstage = isOffstage(opntPosition, stageId);
+  const opntIsDodging = isDodging(oppActionStateId);
+  const opntIsShielding = isShielding(oppActionStateId);
+  const opntIsTeching = isTeching(oppActionStateId);
+  const opntIsDowned = isDown(oppActionStateId);
+  const opntIsShieldBroken = isShieldBroken(oppActionStateId);
+  const opntIsDying = isDead(oppActionStateId);
+  const opntIsLedgeAction = isLedgeAction(oppActionStateId);
 
   // Update percent if opponent didn't lose stock
   if (!opntDidLoseStock) {
     state.conversion.currentPercent = opponentFrame.percent ?? 0;
   }
 
-  if (opntIsDamaged || opntIsGrabbed || opntIsCommandGrabbed) {
+  if (
+    opntIsDamaged ||
+    opntIsGrabbed ||
+    opntIsCommandGrabbed ||
+    opntIsTeching ||
+    opntIsDowned ||
+    opntIsDying ||
+    opntIsOffstage ||
+    opntIsDodging ||
+    opntIsShielding ||
+    opntIsShieldBroken ||
+    opntIsLedgeAction
+  ) {
     // If opponent got grabbed or damaged, reset the reset counter
     state.resetCounter = 0;
-  }
-
-  const shouldStartResetCounter = state.resetCounter === 0 && opntInControl;
-  const shouldContinueResetCounter = state.resetCounter > 0;
-  if (shouldStartResetCounter || shouldContinueResetCounter) {
-    // This will increment the reset timer under the following conditions:
-    // 1) if we were punishing opponent but they have now entered an actionable state
-    // 2) if counter has already started counting meaning opponent has entered actionable state
+  } else {
     state.resetCounter += 1;
   }
 
   let shouldTerminate = false;
 
-  // Termination condition 1 - player kills opponent
-  if (opntDidLoseStock) {
+  // Termination condition 1 - player kills opponent or opponent kills player
+  if (opntDidLoseStock || playerDidLoseStock) {
```

Comment:
`didKill` should not be set to true in the case where the player dies

Reply:
(I apologize for the essay, part of it is leaving breadcrumbs for myself)

Below is a link to a video of g9 top 8 replays parsed by my py-slippi fork (which uses the same logic as this PR) and clippi. I would have used clippi for both but I don't think have the tools/knowledge to build clippi from source without breaking something. Settings were at least 5 hits, at least 60%, must kill (fairly common use-case). I turned all of clippi's bonus features off (chain grab detection, single-hit damage threshold, etc.) since I don't have logic for that in py-slippi yet. They're matched to the frame, so if it only appears on one side, it was captured by one parser and not the other. There'll be some differences in the start/end padding as I wasn't able to match clippi's behavior exactly.

https://youtu.be/osjYLm7PkE0

Most of the combo capture is identical. There are a handful that py-slippi didn't catch, and that looks like it's from a mix of sheik's up B and being airborne above the stage. I think the Y value check is definitely a good idea, but the threshold might require fiddling. 4:35 is a solid false-negative for py-slippi that I need to look more into. The only time I think the combo would have broken is the side B stall -> up B. The shield grab looks a little suspect, but everything that happened afterwards does more than 60% so if there were no other breaks it would have just started there. During the side-B it looks like his X coordinate is beyond the stage, so if I had to guess it's that the fall + drift inwards + up B startup meant he was under the stage 

Py-slippi caught several that weren't caught by slippi-js, and those mainly had to do with holding shield. 5:07 in particular I'm not super happy with, but it's a judgement call. The WDOoS -> dtilt -> shield took 43 frames so it just barely didn't break. Flashing shield in quick succession can lead to unexpected behavior, but it's not 100% wrong either - he's still in a severe disadvantage state from the prior conversion. I've considered increasing the complexity of the logic a tiny bit for instances like these, but I'm not sure how y'all feel about adding more "visible" complexity. For example, in this case I'd make shielding stall the timer and maybe take up to 20 frames off, or reverse the timer by 1-2 frames for every frame it's active, rather than resetting it completely to 0. 

There wasn't any that I'd consider true false-positives from either parser, but I'll try to find some sample replays that have some. I haven't watched top 8 yet so I can't comment on false-negatives that affect both parsers at the moment, but if you noticed some let me know and I'll see what I can do about fixing them.

As an aside, I noticed at 3:08 that the extra padding on clippi resulted in catching some startup shield pressure that was missed in py-slippi. In the future I could make that more intentional and less accidental by looking for hitlag while in shield that occurs within X frames of a conversion starting, then just swapping the start time of the conversion with the shield pressure's start time. That might also be useful for the extending "opening types" stat generation.